src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/MetaGuidanceServiceTests.cs

```diff
@@ -0,0 +1,536 @@
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                await contentDbContext.AddRangeAsync(contentReleaseVersion1, contentReleaseVersion2);
+                await contentDbContext.AddRangeAsync(file1, file2);
+                await contentDbContext.AddRangeAsync(releaseVersion1File1, releaseVersion2File1, releaseVersion2File2);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                await statisticsDbContext.AddRangeAsync(statsReleaseVersion1, statsReleaseVersion2);
+                await statisticsDbContext.AddRangeAsync(subject1, subject2);
+                await statisticsDbContext.AddRangeAsync(releaseVersion1Subject1, releaseVersion2Subject1,
+                    releaseVersion2Subject2);
+                await statisticsDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbContextId))
+            {
+                var service = SetupMetaGuidanceService(contentDbContext: contentDbContext,
+                    statisticsDbContext: statisticsDbContext);
+
+                // Update Subject 1 on version 2
+                var updateResult = await service.UpdateSubject(
+                    contentReleaseVersion2.Id, subject1.Id, new MetaGuidanceUpdateSubjectViewModel
+                    {
+                        Content = "Version 2 Subject 1 Meta Guidance Updated"
+                    });
+
+                // Assert only one Subject has been updated
+
+                Assert.True(updateResult.IsRight);
+
+                Assert.Equal(contentReleaseVersion2.Id, updateResult.Right.Id);
+                Assert.Equal("Version 2 Release Meta Guidance", updateResult.Right.Content);
+                Assert.Equal(2, updateResult.Right.Subjects.Count);
+
+                Assert.Equal(subject1.Id, updateResult.Right.Subjects[0].Id);
+                Assert.Equal("Version 2 Subject 1 Meta Guidance Updated", updateResult.Right.Subjects[0].Content);
+                Assert.Equal("file1.csv", updateResult.Right.Subjects[0].Filename);
+                Assert.Equal("Subject 1", updateResult.Right.Subjects[0].Name);
+
+                Assert.Equal(subject2.Id, updateResult.Right.Subjects[1].Id);
+                Assert.Equal("Version 2 Subject 2 Meta Guidance", updateResult.Right.Subjects[1].Content);
+                Assert.Equal("file2.csv", updateResult.Right.Subjects[1].Filename);
+                Assert.Equal("Subject 2", updateResult.Right.Subjects[1].Name);
+
+                // Assert the same Subject on version 1 hasn't been affected
+                var version1Result = await service.Get(contentReleaseVersion1.Id);
```

Comment:
Again, would split out a separate test just for checking amendments.

Reply:
Cool, have done this.